src/blog/2018-09-24/mutation-switching.pug

```diff
@@ -0,0 +1,116 @@
+        and many many hours are being put in to make it as fast as possible. With the goals of stryker in mind (It's easy to use and fast to run) 
+        we needed to come up with an intelligent way to deal with compilation time.
+        
+        ## Compiling each mutation
+        One way of introducing mutations to a codebase is mutating one statement, compile to code, run tests and gather the results. 
+        This is a logical choice because you have the maximum flexibility and are certain what effect the mutant had. 
+        When looking at the following example there are three mutations possible, we can change the `>` into `>=`, `<` and  `==`.
+        ```scala
+        class Numbers {
+            def greaterThan(a: Int, b: Int): Boolean = {
+                a > b
+            }
+        }
+        ```
+        If we apply the mutation one by one we would need to compile the code base three times. 
+        If we assume the compile time of this program is 10 seconds, we already have 30 seconds of compile time for one full mutation run. 
+        This would quickly get out of hand when the code base is bigger and has more mutations available.
+        
+        ## Mutation switching
+        Mutation switching to the rescue! So how is mutation switching faster than compiling each mutation on its own? 
+        The steps are quite similar but have some minor differences. First, All possible mutations are identified for the whole code base. 
+        Secondly, all mutations are applied to the code base using a [Scala Pattern match](https://docs.scala-lang.org/tour/pattern-matching.html) 
+        with the original satement in the default case, and activated using environment variables.
+        If we look at the same code example as used previously we would get the following code base after the mutations are applied. 
+    
+        ```scala
+        class Numbers {
+            def greaterThan(a: Int, b: Int): Boolean = {  
+                sys.env.get("ACTIVE_MUTATION") match {  
+                    case Some("0") =>  
+                        a >= b  
+                    case Some("1") =>  
+                        a < b  
+                    case Some("2") =>  
+                        a == b  
+                    case _ =>  
+                        a > b  
+                }
+            }  
+        }
+        ```
+        All possible mutations are implemented in the pattern match with an identifier so we can turn on specific mutations 
+        and a default case so the code will work normally one none of the mutants are activated. Now the code base only needs to be compiled once, 
+        but because of the increased size of the code base the compilation time will be a bit longer but that does not way up 
```

Comment:
Suggestion: "but that does not way up against the full recompilation as before" -> "but that is negligible compared to the overhead of compiling each mutant."